gdx/src/com/badlogic/gdx/graphics/g2d/TextureArraySpriteBatch.java

```diff
@@ -213,11 +217,12 @@ public static ShaderProgram createDefaultShader (int maxTextureUnits) {
 			+ "}\n";
 
 		// The texture is simply selected from an array of textures
-		String fragmentShader = "#ifdef GL_ES\n" //
+		String fragmentShader = (Gdx.graphics.isGL30Available() ? "#version 300 es\n" : "#version 100\n")

```

Comment:
not sure about version 100: 
if i'm not wrong, GLSL version starts from 110 (see https://github.com/mattdesl/lwjgl-basics/wiki/GLSL-Versions#glsl-versions)
and i know from experience that MacOS require version 120 (it fails with version 110 suprizingly)

Reply:
I think "300 es" won't work either, as MacOS only supports version 150 for OpenGL 3.x. @VaTTeRGeR Is there even a reason why you're setting the version?

Reply:
@mgsx-dev Thanks for the MacOS info, i'm going to try something different now, i'll report back later.

@crykn The reason is that AMDs GLSL compiler does not seem to bother about throwing an error unless "#version 300 es" is specified, it would otherwise silently accept the fragment-shader and cause visual glitches.

I do not have access to a machine running MacOS, but setting "#version 150" also works for me on Windows so i might run some tests with that setting plus a conditional switch for mobile. I still need to check on the AMD machine if the compiler reports the error correctly as it does with "#version 300 es", if it gets too crazy i'll cancel the pull request or something.